src/sentry/monitors/processing_errors/manager.py

```diff
@@ -98,28 +98,28 @@ def _delete_for_entity(entity_identifier: str, uuid: uuid.UUID) -> None:
     pipeline.execute()
 
 
-def _delete_for_entity_by_type(entity_identifier: str, type: ProcessingErrorType) -> None:
+def _delete_for_entity_by_type(
+    entity_identifier: str, type: ProcessingErrorType, monitor: Monitor | None = None
+) -> None:
     checkin_errors = _get_for_entities([entity_identifier])
     redis = _get_cluster()
     pipeline = redis.pipeline()
     for checkin_error in checkin_errors:
         errors = checkin_error.errors
         if not any(error["type"] == type for error in errors):
             continue
-
-        # If the processing error only holds this one type of error, remove the whole error
-        if len(errors) == 1:
-            pipeline.zrem(build_set_identifier(entity_identifier), checkin_error.id.hex)
-            pipeline.delete(build_error_identifier(checkin_error.id))
-        # If the processing error has other errors, filter out the matching error and update the redis value
-        else:
+        # If the processing error holds more than one type of error, re-store the filtered version
+        # with the target error type removed
+        if len(errors) > 1:
             filtered_errors = list(filter(lambda error: error["type"] != type, errors))
             new_checkin_error = CheckinProcessingError(filtered_errors, checkin_error.checkin)
```

Comment:
Does it make sense here to create the `CheckinProcessingError` with the id from `checkin_error`? I think that might have just been an oversight. Then we don't need to update multiple places